sqld/src/replication/primary/logger.rs

```diff
@@ -897,6 +897,14 @@ fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
             );
             Ok(())
         })?;
+        // turn off auto_checkpointing - we'll use a fiber to checkpoint in time steps instead
+        let rc = rusqlite::ffi::sqlite3_wal_autocheckpoint(conn.handle(), 0);
```

Comment:
I saw later that the autocheckpoint interval is also allowed to be off (it's an Option<u64>). In that case, I suppose we should *not* turn off autocheckpointing, because it would make the database never checkpoint and just go out of storage. And if somebody wants that behavior, they can set the checkpoint interval to 1 billion hours.